provisioning/plugins/corim-psa-decoder/extractor.go

```diff
@@ -26,61 +43,61 @@ func (o Extractor) SwCompExtractor(rv comid.ReferenceValue) ([]*proto.Endorsemen
 	// measurements as needed, provided they belong to the same PSA RoT
 	// identified in the subject of the "reference value" triple.  A single
 	// reference-triple-record SHALL completely describe the updatable PSA RoT.
-	swComponents := make([]*proto.Endorsement, 0, len(rv.Measurements))
-
+	refVals := make([]*proto.Endorsement, 0, len(rv.Measurements))
+	var refVal *proto.Endorsement
+	var err error
 	for i, m := range rv.Measurements {
-		var psaSwCompAttrs PSASwCompAttributes
-
-		if err := psaSwCompAttrs.FromMeasurement(m); err != nil {
-			return nil, fmt.Errorf("extracting measurement at index %d: %w", i, err)
+		if m.Key == nil {
+			return nil, fmt.Errorf("measurement key is not present")
 		}
-
-		swAttrs, err := makeSwAttrs(psaClassAttrs, psaSwCompAttrs)
-		if err != nil {
-			return nil, fmt.Errorf("failed to create software component attributes: %w", err)
+		if !m.Key.IsSet() {
+			return nil, fmt.Errorf("measurement key is not set")
 		}
-
-		swComponent := proto.Endorsement{
-			Scheme:     proto.AttestationFormat_PSA_IOT,
-			Type:       proto.EndorsementType_REFERENCE_VALUE,
-			Attributes: swAttrs,
+		// Check which MKey is present and then decide which extractor to invoke
+		if m.Key.IsPSARefValID() {
+			var psaSwCompAttrs PSASwCompAttributes
+
+			refVal, err = ExtractMeas(&psaSwCompAttrs, m, psaClassAttrs)
+			if err != nil {
+				return nil, fmt.Errorf("unable to extract measurement at index %d, %w", i, err)
+			}
 		}
-
-		swComponents = append(swComponents, &swComponent)
+		if m.Key.IsCCAPlatformConfigID() {
```

Comment:
is this an `else if` ?

Reply:
Yes, I have corrected it. Though I admit, we do not have a method to express of the type of MKey, unless I am missing something!